k2/csrc/array.h

```diff
@@ -224,8 +224,11 @@ class Array1 {
     } else {
       K2_CHECK_EQ(type, kCuda);
       T ans;
-      MemoryCopy(static_cast<void *>(&ans), static_cast<const void *>(data),
-                 ElementSize(), MemcpyDeviceToHost);
+      cudaError_t ret = cudaMemcpy(static_cast<void *>(&ans),
```

Comment:
using `MemoryCopy` instead of `cudaMemcpy`? I think it's better if we can use consistent API for memory copy. (And if we want to  use cudaMemcpy directly, then we should delete MemoryCopy)

Reply:
I wasn't crazy about the idea of wrapping the cudaMemcpyKind in another enum.  I don't like doing these kinds of things before they are actually necessary (i.e. having one-to-one maps like this).
BTW I slightly modified MemoryCopy so it uses asynchronous mode for device-to-device transfers, for speed,
and uses the appropriate stream in that case.